### ISSUE TITLE
Naming clashes

### ISSUE THREAD

**Issue by hellerve** [carp-lang/Carp#195]
When defining variables that clash with typenames, we get spurious error from the C compiler, most notably when naming things `string` or `pattern`—which might be desirable names.

Minimum viable example:
```
(def string "hi")

(defn main []
  (IO.println string))
```

Please note that while `string` and `pattern` are most certainly the most notable examples, this also works with any other C type, e.g. `uint8_t`.

I guess this could be fixed by mangling variable names as well?

Cheers

**Comment by eriksvedang** [comment]
Right, this is annoying. Changing the C names to `String` and `Pattern` seems like a good and simple start. Perhaps preventing identifiers from starting with a capital letter could be worthwhile. Or at least having a list of restricted words for symbols.

**Comment by hellerve** [comment]
That sounds sensible enough for the moment!

**Comment by hellerve** [comment]
Can this be closed as per #200?
